homeassistant/components/androidtv/media_player.py

```diff
@@ -176,51 +176,30 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         _LOGGER.warning("Platform already setup on %s, skipping", address)
         return
 
+    adbkey = config.get(CONF_ADBKEY, hass.config.path(STORAGE_DIR, "androidtv_adbkey"))
     if CONF_ADB_SERVER_IP not in config:
         # Use "adb_shell" (Python ADB implementation)
-        if CONF_ADBKEY not in config:
-            # Generate ADB key files (if they don't exist)
-            adbkey = hass.config.path(STORAGE_DIR, "androidtv_adbkey")
-            if not os.path.isfile(adbkey):
-                keygen(adbkey)
-
-            adb_log = f"using Python ADB implementation with adbkey='{adbkey}'"
-
-            aftv = setup(
-                config[CONF_HOST],
-                config[CONF_PORT],
-                adbkey,
-                device_class=config[CONF_DEVICE_CLASS],
-                state_detection_rules=config[CONF_STATE_DETECTION_RULES],
-                auth_timeout_s=10.0,
-            )
+        if not os.path.isfile(adbkey):
```

Comment:
This should be run in executor too.

Reply:
You would want to wrap the whole part of `adbkey = …` till `aftv = …` in a function and pass that to the executor.

Reply:
Are you saying that these lines should not be run in the event loop:

```python
adbkey = config.get(CONF_ADBKEY, hass.config.path(STORAGE_DIR, "androidtv_adbkey"))

if not os.path.isfile(adbkey):
```

Reply:
`os.path.isfile` is doing I/O as it looks on the hard drive if something is there.

Reply:
But since it's inefficient to go to the executor multiple times, you can just create a function and run a whole block of code in the executor.

```python
def sync_work():
    adbkey = config.get(CONF_ADBKEY, hass.config.path(STORAGE_DIR, "androidtv_adbkey"))

    if not os.path.isfile(adbkey):
        return False

    return True

success = await hass.async_add_executor_job(sync_work)
```

Reply:
I put all of the I/O in `async_setup_platform` into a single executor job.